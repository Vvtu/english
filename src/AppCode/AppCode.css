:root {
	--font-size: 24px;

	--overlap_bottom: 8px;
	--info_panel_height: calc(var(--font-size)*2);
	--main_gap: calc(var(--font-size) / 2);

	--text-height-portrait: calc(
		 (100vh - var(--overlap_bottom) - (var(--main_gap) * 2) - var(--info_panel_height)) /
			2
	);

	--text-height-landscape: calc(
		100vh - var(--info_panel_height) - var(--main_gap) - var(--overlap_bottom)
	);

	--advanced-color: #522719;
	--background-color: #000000;
	--base-text-color: #ffffff;
	--danger-color: #ff4000;
	--english-appcode__speaking-text-color: #6b5ee0;
	--english-text-color: #94a11f;
	--grey-color: #777777;
	--shadow-color: #dddddd;
}

/* grid
============================================================= */

.appcode__grid {
	background-color: var(--background-color);
	color: var(--base-text-color);
	display: grid;
	filter: invert(100%);
	font-size: var(--font-size);
	grid-column-gap: var(--main_gap);
	grid-row-gap: var(--main_gap);
	padding: 0 8px;
}

@media (orientation: portrait) {
	.appcode__grid {
		grid-template-columns: 1fr;
		grid-template-rows: var(--text-height-portrait) var(--info_panel_height) var(--text-height-portrait);
		grid-template-areas: 'russian' 'info' 'english';
	}
	.appcode__icon_rotate {
		transform: rotate(270deg);
	}
	.appcode__icon_rotate_back {
		transform: rotate(90deg);
	}
}

@media (orientation: landscape) {
	.appcode__grid {
		grid-template-columns: 1fr 1fr;
		grid-template-rows: var(--text-height-landscape) var(--info_panel_height);
		grid-template-areas: 'russian english' 'info info';
	}
	.appcode__icon_rotate {
		transform: rotate(90deg);
	}
	.appcode__icon_rotate_back {
		transform: rotate(270deg);
	}
}

.appcode__info {
	grid-area: info;
}

.appcode__russian {
	grid-area: russian;
}

.appcode__english {
	grid-area: english;
}

.appcode__scroll {
	max-height: 100%;
	overflow-y: scroll;
	-webkit-overflow-scrolling: touch;
}

/* font size and colors
============================================================= */

.appcode__eng_text_color {
	color: var(--english-text-color);
}

.appcode__speaking {
	color: var(--english-appcode__speaking-text-color);
}

/* icons
============================================================= */

.appcode__icon_invert__horizontal {
	transform: scaleX(-1);
}

/* flex boxes
============================================================= */

.appcode__center {
	height: 100%;
	display: flex;
	flex-direction: row;
	justify-content: center;
	align-items: center;
}

.appcode__info2 {
	height: 100%;
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	align-items: center;
}
