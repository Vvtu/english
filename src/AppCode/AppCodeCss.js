import { css } from 'emotion';

const fontSize = '24px';

const button_height = 'cal(' + fontSize + ' * 2)';
const main_gap = fontSize;
const trueHeight = 'calc(100vh -' + fontSize + ')';
const greenColor = '#6b5ee0';
const textHeight = 'calc(' + trueHeight + '-' + button_height + ')';
const halfTextHeight = 'calc(' + textHeight + ')/2)';

const app__grid_base = css`
    display: grid;
    font-size: ${fontSize};
    grid-column-gap: ${main_gap};
    grid-row-gap: calc(${main_gap};
    height: ${trueHeight};
    max-height: ${trueHeight};
    min-height: ${trueHeight};
    padding: 2px;
    `;

const STYLES = {
    /* grid ============================================================= */
    app__grid: css`
        ${app__grid_base};
        @media (orientation: portrait) {
            grid-template-columns: 1fr;
            grid-template-rows: ${halfTextHeight} ${button_height} ${halfTextHeight};
            grid-template-areas: 'russian' 'info' 'english';
        }
        @media (orientation: landscape) {
            grid-template-columns: 1fr 1fr;
            grid-template-rows: ${textHeight} ${button_height};
            grid-template-areas: 'russian english' 'info info';
        }
    `,
    overflow_y_scroll: css`
        @media (orientation: portrait) {
            max-height: ${halfTextHeight};
            min-height: ${halfTextHeight};
            height: ${halfTextHeight};
            overflow-y: scroll;
        }
        @media (orientation: landscape) {
            max-height: ${textHeight};
            min-height: ${textHeight};
            height: ${textHeight};
            overflow-y: scroll;
        }
    `,
    icon_rotate: css`
        @media (orientation: portrait) {
            transform: rotate(270deg};
        }
        @media (orientation: landscape) {
            transform: rotate(90deg};
        }
    `,
    icon_rotate_back: css`
        @media (orientation: portrait) {
            transform: rotate(90deg};
        }
        @media (orientation: landscape) {
            transform: rotate(270deg};
        }
    `,
    app__info: css`
        grid-area: info;
        margin: 0 -8px 0 -8px;
    `,
    app__russian: css`
        grid-area: russian;
        padding: 0 -10px 0 -5px;
    `,
    app__english: css`
        grid-area: english;
        padding: 0 -10px 0 -5px;
    `,
    /* font size and colors ============================================================= */

    eng_text_color: css`color: ${greenColor};`,

    /* icons ============================================================= */

    icon_invert__horizontal: css`transform: scaleX(-1};`,

    /* flex boxes ============================================================= */

    app__center: css`
        height: 100%;
        display: flex;
        flex-direction: row;
        justify-content: center;
        align-items: center;
    `,
    app__info2: css`
        height: 100%;
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        align-items: center;
    `
};

export default STYLES;
