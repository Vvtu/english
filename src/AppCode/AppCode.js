import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import PopupWindowForAdvancedMenu from 'PopupWindowForAdvancedMenu/PopupWindowForAdvancedMenu';
import ArrowIcon from 'AppCode/svg/ArrowIcon';
import AdvancedIcon from 'AppCode/svg/AdvancedIcon';

import 'AppCode/AppCode.css';

import { getDictionaryWithMix, handleDictClicked } from 'lib/lib';

console.log('process.env = ', process.env);
console.log(
	'process.env.REACT_APP_SECRET_CODE = ',
	process.env && process.env.REACT_APP_SECRET_CODE,
);

class AppCode extends PureComponent {
	static propTypes = {
		dictionaries: PropTypes.object.isRequired,
	};
	constructor(props) {
		super(props);
		this.state = {
			activeIndex: 0,
			showEnglish: false,
			showAdvanced: false,
			randomDictionary: getDictionaryWithMix(this.props.dictionaries),
		};
	}

	componentDidMount() {
		// window.addEventListener('touchstart', this.handleTouchStart);
		const styles = getComputedStyle(document.documentElement);
		this.setState({
			greenColor: String(styles.getPropertyValue('--english-text-color')).trim(),
			advancedColor: String(styles.getPropertyValue('--advanced-color')).trim(),
			whiteColor: String(styles.getPropertyValue('--base-text-color')).trim(),
		});
	}

	// componentWillUnmount() {
	// 	window.removeEventListener('touchstart', this.handleTouchStart);
	// }

	// handleTouchStart = (e) => {
	// 	console.log('e = ', e);
	// 	e.preventDefault();
	// 	console.log('e = ', e);
	// 	return false;
	// };

	incrementLocalStorage = () => {
		const { activeIndex, randomDictionary, showEnglish } = this.state;
		if (showEnglish) {
			const activeObj = activeIndex !== undefined && randomDictionary[activeIndex];
			const russian = Object.keys(activeObj || {})[0];
			if (russian) {
				const item = localStorage.getItem(russian);
				const shown = parseInt(item, 10) || 0;
				const newItem = String(shown + 1);
				localStorage.setItem(russian, newItem);
			}
		}
	};

	handleForwardClicked = (e) => {
		e.preventDefault();
		e.stopPropagation();
		this.incrementLocalStorage();

		const { activeIndex, randomDictionary } = this.state;

		const len = randomDictionary.length;
		const newActiveIndex = activeIndex === len - 1 ? 0 : activeIndex + 1;
		console.log('handleForwardClicked newActiveIndex = ', newActiveIndex);
		this.setState({
			activeIndex: newActiveIndex,
			showEnglish: false,
		});
		return false;
	};

	handleBackClicked = (e) => {
		e.preventDefault();
		e.stopPropagation();
		this.incrementLocalStorage();

		const { activeIndex, randomDictionary } = this.state;
		const len = randomDictionary.length;
		const newActiveIndex = activeIndex === 0 ? len - 1 : activeIndex - 1;
		console.log('handleBackClicked newActiveIndex = ', newActiveIndex);
		this.setState({
			activeIndex: newActiveIndex,
			showEnglish: false,
		});
		return false;
	};

	handleShowEnglishClicked = (e) => {
		e.preventDefault();
		e.stopPropagation();
		this.setState({
			showEnglish: true,
		});
		return false;
	};

	handleAdvancedClicked = (e) => {
		e.preventDefault();
		e.stopPropagation();
		this.setState({
			showAdvanced: true,
		});
		return false;
	};
	handleClosePopupClicked = () => {
		this.setState({
			showAdvanced: false,
		});
		return false;
	};

	handleHideItemClicked = () => {
		const { activeIndex, randomDictionary } = this.state;
		if (activeIndex !== undefined && activeIndex < randomDictionary.length) {
			const activeObj = randomDictionary[activeIndex];
			const russian = Object.keys(activeObj || {})[0];
			if (russian) {
				localStorage.setItem(russian, 'd');
			}
			const newRandomDictionary = randomDictionary
				.slice(0, activeIndex)
				.concat(randomDictionary.slice(activeIndex + 1));
			const newActiveIndex = activeIndex >= newRandomDictionary.length ? 0 : activeIndex;
			this.setState({
				activeIndex: newActiveIndex,
				randomDictionary: newRandomDictionary,
				showEnglish: false,
				showAdvanced: false,
			});
		}
	};

	handleStatisticClicked = (e) => {
		e.preventDefault();
		e.stopPropagation();
		if (this.state.showEnglish) {
			this.handleForwardClicked(e);
		} else {
			this.handleShowEnglishClicked(e);
		}
		return false;
	};

	handleUnhideAllItemsClicked = () => {
		localStorage.clear();
		this.setState({
			activeIndex: 0,
			randomDictionary: getDictionaryWithMix(this.props.dictionaries),
			showEnglish: false,
			showAdvanced: false,
		});
	};

	handleTextToSpeachClicked = (e, text) => {
		e.preventDefault();
		e.stopPropagation();
		const utterThis = new SpeechSynthesisUtterance();
		const synth = window.speechSynthesis;
		if (utterThis && synth) {
			this.setState({ speaking: true });
			// const voices = synth.getVoices();
			// utterThis.voice = voices[1];   // ошибка какая то !!!
			// utterThis.voiceURI = 'Google UK English Male';
			// utterThis.volume = 1; // 0 to 1
			// utterThis.rate = 0.1; // 0.1 to 10
			// utterThis.pitch = 0; //0 to 2
			utterThis.lang = 'en-GB';
			utterThis.onend = (event) => {
				this.setState({ speaking: false });
			};
			utterThis.text = text;
			synth.speak(utterThis);
		}
		return false;
	};

	handleDictClickedLocal = (number) => {
		handleDictClicked(number);
		this.setState({
			activeIndex: 0,
			randomDictionary: getDictionaryWithMix(this.props.dictionaries),
			showEnglish: false,
			showAdvanced: false,
		});
	};

	render() {
		const {
			activeIndex,
			randomDictionary,
			showAdvanced,
			showEnglish,
			speaking,
			greenColor,
			advancedColor,
			whiteColor
		} = this.state;

		const activeObj = activeIndex !== undefined && randomDictionary[activeIndex];
		const russian = Object.keys(activeObj || {})[0];

		const english = Object.values(activeObj || {})[0];
		const count = randomDictionary.length;

		const item = localStorage.getItem(russian);
		const shown = parseInt(item, 10) || 0;

		return (
			<div>
				<div className="app__grid">
					<div className="app__info">
						<div className="app__info2">
							<ArrowIcon
								height={32}
								onClick={this.handleBackClicked}
								onDoubleClick={this.handleBackClicked}
								width={32}
								fill={whiteColor}
							/>
							<AdvancedIcon
								height={32}
								onClick={this.handleAdvancedClicked}
								onDoubleClick={this.handleAdvancedClicked}
								width={32}
								fill={advancedColor}
							/>

							{!showEnglish && (
								<ArrowIcon
									className="icon_rotate"
									height={32}
									onClick={this.handleShowEnglishClicked}
									onDoubleClick={this.handleShowEnglishClicked}
									width={32}
									fill={greenColor}
								/>
							)}
							{showEnglish && (
								<ArrowIcon
									className="icon_rotate_back"
									height={32}
									onClick={this.handleForwardClicked}
									onDoubleClick={this.handleForwardClicked}
									width={32}
									fill={greenColor}
								/>
							)}

							<div
								onClick={this.handleStatisticClicked}
								onDoubleClick={this.handleStatisticClicked}
							>
								<span>{activeIndex + 1 + '/' + count}</span>
								<span className="eng_text_color">{'(' + shown + ')'}</span>
							</div>

							<ArrowIcon
								className="icon_invert__horizontal"
								height={32}
								onClick={this.handleForwardClicked}
								onDoubleClick={this.handleForwardClicked}
								width={32}
								fill={whiteColor}
							/>
						</div>
					</div>
					<div className="app__russian">
						<div className="app__center">
							<div className="overflow_y_scroll">
								<div>{russian}</div>
							</div>
						</div>
					</div>
					<div
						className="app__english"
						onClick={(e) => this.handleTextToSpeachClicked(e, english)}
					>
						<div className="app__center">
							<div className="overflow_y_scroll">
								<div className={'eng_text_color' + (speaking ? ' speaking' : '')}>
									{showEnglish && english}
								</div>
							</div>
						</div>
					</div>

					{showAdvanced && (
						<PopupWindowForAdvancedMenu
							handleClosePopupClicked={this.handleClosePopupClicked}
							handleHideItemClicked={this.handleHideItemClicked}
							handleUnhideAllItemsClicked={this.handleUnhideAllItemsClicked}
							handleDictClicked={this.handleDictClickedLocal}
						/>
					)}
				</div>
			</div>
		);
	}
}

export default AppCode;
