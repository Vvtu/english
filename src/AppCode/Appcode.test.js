import React from 'react';
import renderer from 'react-test-renderer';

import AppCode from './AppCode';

const dictionary = [ { 'Русский текст': 'Russian text' } ];

describe('AppCode test   ', () => {
    it('test 1', () => {
        const component = renderer.create(<AppCode dictionary={dictionary} />);
        const result = component.toJSON();
        expect(result).toMatchSnapshot();
    });
});
