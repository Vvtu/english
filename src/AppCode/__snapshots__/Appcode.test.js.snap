// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppCode test    test 1 1`] = `
<div>
  <div
    className="app__grid"
  >
    <div
      className="app__info"
    >
      <div
        className="app__info2"
      >
        <svg
          fill="#00000"
          height={32}
          onClick={[Function]}
          onDoubleClick={[Function]}
          viewBox="0 0 141.732 141.732"
          width={32}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M105.614,118.681c3.398,3.396,3.4,8.912,0,12.311c-3.396,3.399-8.91,3.398-12.311,0c-0.02-0.02-0.035-0.04-0.053-0.061 l-0.025,0.022l-57.66-57.66l0.024-0.022c-1.61-1.579-2.608-3.775-2.608-6.208c-0.002-2.73,1.258-5.166,3.229-6.762l-0.06-0.058 l57.66-57.66l0.025,0.024c0.018-0.021,0.033-0.039,0.053-0.058c3.4-3.4,8.912-3.4,12.312,0c3.398,3.396,3.398,8.908,0,12.311 c-0.021,0.02-0.041,0.034-0.061,0.054l0.023,0.024L54.043,67.063l51.54,51.54l-0.025,0.021 C105.573,118.646,105.594,118.66,105.614,118.681"
          />
        </svg>
        <svg
          fill="#add8e6"
          height={32}
          onClick={[Function]}
          onDoubleClick={[Function]}
          viewBox="0 0 24 24"
          width={32}
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="icons"
          >
            <path
              d="M22.2,14.4L21,13.7c-1.3-0.8-1.3-2.7,0-3.5l1.2-0.7c1-0.6,1.3-1.8,0.7-2.7l-1-1.7c-0.6-1-1.8-1.3-2.7-0.7 L18,5.1c-1.3,0.8-3-0.2-3-1.7V2c0-1.1-0.9-2-2-2h-2C9.9,0,9,0.9,9,2v1.3c0,1.5-1.7,2.5-3,1.7L4.8,4.4c-1-0.6-2.2-0.2-2.7,0.7 l-1,1.7C0.6,7.8,0.9,9,1.8,9.6L3,10.3C4.3,11,4.3,13,3,13.7l-1.2,0.7c-1,0.6-1.3,1.8-0.7,2.7l1,1.7c0.6,1,1.8,1.3,2.7,0.7L6,18.9 c1.3-0.8,3,0.2,3,1.7V22c0,1.1,0.9,2,2,2h2c1.1,0,2-0.9,2-2v-1.3c0-1.5,1.7-2.5,3-1.7l1.2,0.7c1,0.6,2.2,0.2,2.7-0.7l1-1.7 C23.4,16.2,23.1,15,22.2,14.4z M12,16c-2.2,0-4-1.8-4-4c0-2.2,1.8-4,4-4s4,1.8,4,4C16,14.2,14.2,16,12,16z"
              id="settings"
            />
          </g>
        </svg>
        <svg
          className="icon_rotate"
          fill="#6b5ee0"
          height={32}
          onClick={[Function]}
          onDoubleClick={[Function]}
          viewBox="0 0 141.732 141.732"
          width={32}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M105.614,118.681c3.398,3.396,3.4,8.912,0,12.311c-3.396,3.399-8.91,3.398-12.311,0c-0.02-0.02-0.035-0.04-0.053-0.061 l-0.025,0.022l-57.66-57.66l0.024-0.022c-1.61-1.579-2.608-3.775-2.608-6.208c-0.002-2.73,1.258-5.166,3.229-6.762l-0.06-0.058 l57.66-57.66l0.025,0.024c0.018-0.021,0.033-0.039,0.053-0.058c3.4-3.4,8.912-3.4,12.312,0c3.398,3.396,3.398,8.908,0,12.311 c-0.021,0.02-0.041,0.034-0.061,0.054l0.023,0.024L54.043,67.063l51.54,51.54l-0.025,0.021 C105.573,118.646,105.594,118.66,105.614,118.681"
          />
        </svg>
        <div
          onClick={[Function]}
          onDoubleClick={[Function]}
        >
          <span>
            1/1
          </span>
          <span
            className="eng_text_color"
          >
            (0)
          </span>
        </div>
        <svg
          className="icon_invert__horizontal"
          fill="#00000"
          height={32}
          onClick={[Function]}
          onDoubleClick={[Function]}
          viewBox="0 0 141.732 141.732"
          width={32}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M105.614,118.681c3.398,3.396,3.4,8.912,0,12.311c-3.396,3.399-8.91,3.398-12.311,0c-0.02-0.02-0.035-0.04-0.053-0.061 l-0.025,0.022l-57.66-57.66l0.024-0.022c-1.61-1.579-2.608-3.775-2.608-6.208c-0.002-2.73,1.258-5.166,3.229-6.762l-0.06-0.058 l57.66-57.66l0.025,0.024c0.018-0.021,0.033-0.039,0.053-0.058c3.4-3.4,8.912-3.4,12.312,0c3.398,3.396,3.398,8.908,0,12.311 c-0.021,0.02-0.041,0.034-0.061,0.054l0.023,0.024L54.043,67.063l51.54,51.54l-0.025,0.021 C105.573,118.646,105.594,118.66,105.614,118.681"
          />
        </svg>
      </div>
    </div>
    <div
      className="app__russian"
    >
      <div
        className="overflow_y_scroll"
      >
        <div
          className="app__center"
        >
          Русский текст
        </div>
      </div>
    </div>
    <div
      className="app__english"
    >
      <div
        className="overflow_y_scroll"
      >
        <div
          className="app__center eng_text_color"
        />
      </div>
    </div>
  </div>
</div>
`;
